File: manual/code/generate_strategies/IR(pseudocode)/python/python/function_collections_ordereddict.py
from collections import OrderedDict


class OrderedDictCalculator:
    def process_key_value_pairs(self, data: str) -> str:
        if not data:
            return "failed"

        ordered_dict = OrderedDict()

        items = data.split(",")
        for item in items:

            if item.strip():
                try:
                    key, value = item.split(":")
                    ordered_dict[key.strip()] = value.strip()
                except ValueError:
                    continue

        if not ordered_dict:
            return "failed"

        result = ', '.join(f"{key}:{value}" for key, value in ordered_dict.items())
        return result

File: manual/code/test_suites/cpp/FunctionBuildTest/src/function_collections_ordereddict.cpp
#include <string>
#include <vector>
#include <map>
#include <sstream>
#include <algorithm>

class OrderedDictCalculator {
public:
    std::string process_key_value_pairs(const std::string& data) {
        if (data.empty()) {
            return "failed";
        }

        std::map<std::string, std::string> orderedDict;
        std::vector<std::string> items;
        std::stringstream ss(data);
        std::string item;

        while (std::getline(ss, item, ',')) {
            item.erase(0, item.find_first_not_of(' '));
            item.erase(item.find_last_not_of(' ') + 1);
            if (item.empty()) {
                continue;
            }

            size_t colonPos = item.find(':');
            if (colonPos == std::string::npos) {
                continue;
            }

            std::string key = item.substr(0, colonPos);
            key.erase(0, key.find_first_not_of(' '));
            key.erase(key.find_last_not_of(' ') + 1);

            std::string value = item.substr(colonPos + 1);
            value.erase(0, value.find_first_not_of(' '));
            value.erase(value.find_last_not_of(' ') + 1);

            orderedDict[key] = value;
        }

        if (orderedDict.empty()) {
            return "failed";
        }

        std::stringstream result;
        bool first = true;
        for (const auto& entry : orderedDict) {
            if (!first) {
                result << ", ";
            }
            first = false;
            result << entry.first << ":" << entry.second;
        }

        return result.str();
    }
};